### ISSUE TITLE
Patching the validate_parameters decorator

### ISSUE THREAD

**Issue by fabiotanaka** [allrod5/parameters-validation#11]
### Problem description
With the actual state of the library it is not intuitive how to write tests for methods that use the validate_parameters decorator. Because decorators are called at the moment that the python module is imported, to patch the decorator we need to disrupt the usual "good programming guidelines" and import the module in the middle of the code after patching it.

#### Disable the type checking
Since the decorator doesn't change the value of the inputs, every testing needed for the validation part can be done separately. So probably just having a way to deactivate the validate_parameters when running tests is good enough for most cases.

#### Postpone the evaluation
I took a brief look at a form of postponing the decorator evaluation so we could patch the method the usual way. Looks like the Python community is aware of the problems that the evaluation of annotations at function definition time can cause since I fond a [PEP for this](https://www.python.org/dev/peps/pep-0563/) discussing that.

**Comment by roo-oliv** [comment]
Hi @fabiotanaka, thanks for being clear and thorough in stating your problem.

I agree that mocking and patching support is not there yet. Though most common tests can live just fine with actual validations being performed I can see where this becomes a problem: if the test validates that some string matches to a real and valid path in your system this is something that you might want to mock during tests.

I'm currently working on a way to solve it.

**Comment by roo-oliv** [comment]
@fabiotanaka two new features were released in version 1.2.0 ([already available at PyPI](https://pypi.org/project/parameters-validation/)) to address this problem.

In short, validations can now be skipped with `.skip_validations()`:

```python
from parameters_validation import non_blank

@validate_parameters
def foo(s: non_blank(str)):
    pass

foo.skip_validations()("")  # does not throw since validations are skipped
```

and validations can now be **mocked** for testing purposes with `.mock_validations({...})`:

```python
from parameters_validation import non_blank

@validate_parameters
def foo(s: non_blank(str)):
    pass

foo.mock_validations({"s": lambda *_: print("mocked")})("")  # prints "mocked"
``` 

Thanks for your help on this!

**Comment by fabiotanaka** [comment]
That's awesome! Thanks for the very quick response and solution =)
